File: dotproduct.rb
# http://rosettacode.org/wiki/Dot_product
# http://en.wikipedia.org/wiki/Dot_product

def dot_product(one,two)
  Array.new(one.count,nil).each_with_index.map {|x,i| x = one[i] * two[i]}.inject(:+)
end

test_array1 = [1,3,-5]
test_array2 = [4,-2,-1]

puts dot_product(test_array1,test_array2)

File: dotproduct.swift
// http://rosettacode.org/wiki/Dot_product

func dotProduct(arrayOne:[Int],arrayTwo:[Int]) -> Int {
  var dotProductArray:[Int] = []
  for var i = 0; i < arrayOne.count; i++ {
    let product = arrayOne[i] * arrayTwo[i]
    dotProductArray.append(product)
  }
  return sumArray(dotProductArray,0)
}

func sumArray(array:[Int],index:Int) -> Int {
  if index >= array.count {
    return 0
  } else {
    return array[index] + sumArray(array,index+1)
  }
}

let testArrayOne = [1,3,-5]
let testArrayTwo = [4,-2,-1]
println(dotProduct(testArrayOne,testArrayTwo))
